modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2073,18 +2073,29 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 				push_error(vformat(R"(Cannot find constant "%s" on type "%s".)", name, base.to_string()), p_identifier);
 			}
 		} else {
-			Callable::CallError temp;
-			Variant dummy = Variant::construct(base.builtin_type, nullptr, 0, temp);
-			List<PropertyInfo> properties;
-			dummy.get_property_list(&properties);
-			for (const List<PropertyInfo>::Element *E = properties.front(); E != nullptr; E = E->next()) {
-				const PropertyInfo &prop = E->get();
-				if (prop.name == name) {
-					p_identifier->set_datatype(type_from_property(prop));
+			switch (base.builtin_type) {
+				case Variant::NIL: // May not needed since it'll be GDScriptParser::DataType::VARIANT now.
```

Comment:
You cannot use indexing on `null` so this case should raise an error I think. Though it should still return as a Variant type so the analyzer can keep checking the rest of the code.

Reply:
I just noticed it is the same 3.2. Still, I think if we inferred the type as `null` we can just raise an error to inform the user.

Reply:
@vnen done! only if a variable is explicitly assigned to `null` the inferred type is `null` and it raise an error otherwise it's a variant.

```gdscript
func _ready():
    var v1 = null
    v1.name ## <-- error
    var v2
    v2.name ## <-- not error
```